### PR TITLE
KAFKA-7141: ConsumerGroupCommand should describe group assignment eve…

### PR DESCRIPTION
**Pull request by huxihx**
…n with no offsets  committed.

https://issues.apache.org/jira/browse/KAFKA-7141

Currently, if a consumer group never commits offsets, ConsumerGroupCommand cannot describe it at all even if the member assignment is valid. Instead, the tool should be able to describe the group information showing empty current_offset and LAG.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by huxihx** [comment]
@vahidhashemian Please review this patch. Thanks.

**Comment by huxihx** [comment]
retest it please.

File: core/src/test/scala/unit/kafka/admin/ConsumerGroupCommandTest.scala [hunk 1]
```diff
@@ -113,8 +114,8 @@ class ConsumerGroupCommandTest extends KafkaServerTestHarness {
 
 object ConsumerGroupCommandTest {
 
-  abstract class AbstractConsumerRunnable(broker: String, groupId: String) extends Runnable {
-    val props = new Properties
+  abstract class AbstractConsumerRunnable(broker: String, groupId: String, customPropsOpt: Option[Properties] = None) extends Runnable {
+    val props = if (customPropsOpt.isDefined) new Properties(customPropsOpt.get) else new Properties
```
Comment (vahidhashemian) [inline core/src/test/scala/unit/kafka/admin/ConsumerGroupCommandTest.scala]: Wouldn't it be better to move this line after `configure(props)` below so we also provide a way to override the default configs? In other words, this line would remain unchanged, and after `configure(props)` we add
```
    if (customPropsOpt.isDefined)
      props.putAll(customPropsOpt.get)
```

**Review by vahidhashemian — APPROVED** [review]
@huxihx thanks for making this improvement. I just had a minor comment inline. Otherwise, LGTM.

**Comment by huxihx** [comment]
@vahidhashemian Thanks for the comments, please review again. 
@hachikuji Please review.

**Comment by vahidhashemian** [comment]
LGTM, thanks for addressing my comment.

**Review by harshach — APPROVED** [review]
+1

**Comment by huxihx** [comment]
@hachikuji Could you take some time to review this patch? Thanks.

**Comment by hachikuji** [comment]
With Vahid's previous KIP, I think the intention is that the default behavior of `--describe` is the same as `--describe --offsets`, so I'm wondering if the behavior is actually reasonable. We also have `--describe --members` which can be used to see the members in the group. Thoughts?

**Comment by vahidhashemian** [comment]
I think this PR fixes the situation where consumers have joined the group but have not committed any offset yet. Current implementation does not report any rows until they commit offsets, which I think does not provide a good user experience. It seems to me that this PR still follows the definition of `--describe` and `--describe --offsets` in [KIP-175](https://cwiki.apache.org/confluence/display/KAFKA/KIP-175%3A+Additional+%27--describe%27+views+for+ConsumerGroupCommand)

> The output of `--describe` will change to return one row per topic partition in the group. This means that group members with no assigned partition will not be present in the default output.

Please advise if I'm missing something.

**Comment by hachikuji** [comment]
Ok, that makes sense.

File: core/src/test/scala/unit/kafka/admin/ConsumerGroupCommandTest.scala [hunk 2]
```diff
@@ -113,9 +114,11 @@ class ConsumerGroupCommandTest extends KafkaServerTestHarness {
 
 object ConsumerGroupCommandTest {
 
-  abstract class AbstractConsumerRunnable(broker: String, groupId: String) extends Runnable {
+  abstract class AbstractConsumerRunnable(broker: String, groupId: String, customPropsOpt: Option[Properties] = None) extends Runnable {
     val props = new Properties
     configure(props)
+    if (customPropsOpt.isDefined)
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/admin/ConsumerGroupCommandTest.scala]: nit: use `foreach`

**Review by hachikuji — APPROVED** [review]
Left a trivial comment, but otherwise LGTM. Thanks!

**Comment by huxihx** [comment]
@hachikuji Thanks for the comments. Please review again.

**Comment by huxihx** [comment]
retest it please.

**Review by hachikuji — APPROVED** [review]
LGTM

**Comment by hachikuji** [comment]
retest this please

**Comment by hachikuji** [comment]
Thanks @huxihx. I will merge in the morning after the build finishes.

**Comment by hachikuji** [comment]
retest this please

**Comment by hachikuji** [comment]
retest this please

**Comment by asfgit** [comment]
FAILURE
 9284 tests run, 5 skipped, 3 failed.
--none--

**Comment by hachikuji** [comment]
@huxihx The build failure on jdk10 seems deterministic. Can you look into it?

**Comment by huxihx** [comment]
retest this please

**Comment by huxihx** [comment]
@hachikuji Seems it hit a known Scala bug opened by Ijuma (https://github.com/scala/bug/issues/10418). I will take care of it.

**Comment by huxihx** [comment]
retest this please

**Comment by hachikuji** [comment]
@huxihx Thanks for fixing the issue. Merged to trunk.

**Pull request merged**